src/main/java/uk/gov/hmcts/darts/authentication/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.hmcts.darts.authentication.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.StreamUtils;
+import uk.gov.hmcts.darts.authentication.config.AuthenticationConfiguration;
+import uk.gov.hmcts.darts.authentication.model.OAuthProviderRawResponse;
+import uk.gov.hmcts.darts.authentication.service.AuthenticationService;
+import uk.gov.hmcts.darts.authentication.service.AzureActiveDirectoryB2CClient;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+@Slf4j
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final Logger logger = LoggerFactory.getLogger(AuthenticationServiceImpl.class);
+
+    @Autowired
+    private AuthenticationConfiguration authServiceConfiguration;
+
+    @Autowired
+    private AzureActiveDirectoryB2CClient azureActiveDirectoryB2CClient;
+
+    @Override
+    @SneakyThrows
+    public String getAuthorizationUrl() {
+        URIBuilder uriBuilder = new URIBuilder(
+            authServiceConfiguration.getExternalADauthorizationUri());
+        uriBuilder.addParameter("client_id", authServiceConfiguration.getExternalADclientId());
+        uriBuilder.addParameter("response_type", authServiceConfiguration.getExternalADresponseType());
+        uriBuilder.addParameter(
+            "redirect_uri",
```

Comment:
Why do we specify the Parameters that are not Required e.g. redirect_uri / response_mode / prompt?

Looking at https://learn.microsoft.com/en-gb/azure/active-directory-b2c/implicit-flow-single-page-application

Wondering are we missing `nonce`?

Reply:
OK thanks Hemanta, I guess it must be me misunderstanding the Azure AD B2C documentation.